internal/forwarder/buffnetlink/server.go

```diff
@@ -0,0 +1,103 @@
+package buffnetlink
+
+import (
+	"encoding/binary"
+	"sync"
+	"syscall"
+
+	"github.com/free5gc/go-gtp5gnl"
+	"github.com/free5gc/go-upf/internal/report"
+	"github.com/khirono/go-genl"
+	"github.com/khirono/go-nl"
+	"github.com/pkg/errors"
+)
+
+type Server struct {
+	client  *nl.Client
+	mux     *nl.Mux
+	conn    *nl.Conn
+	handler report.Handler
+}
+
+var native binary.ByteOrder = gtp5gnl.NativeEndian()
+
+func OpenServer(wg *sync.WaitGroup, client *nl.Client, mux *nl.Mux) (*Server, error) {
+	s := &Server{
+		client: client,
+		mux:    mux,
+	}
+
+	f, err := genl.GetFamily(s.client, "gtp5g")
+	if err != nil {
+		return nil, errors.Wrap(err, "get family")
+	}
+
+	s.conn, err = nl.Open(syscall.NETLINK_GENERIC, int(f.Groups[gtp5gnl.GENL_MCGRP].ID))
+	if err != nil {
+		return nil, errors.Wrap(err, "open netlink")
+	}
+
+	err = s.mux.PushHandler(s.conn, s)
+	if err != nil {
+		return nil, errors.Wrap(err, "push handler")
+	}
+
+	// wg.Add(1)
+	return s, nil
+}
+
+func (s *Server) Close() {
+	s.mux.PopHandler(s.conn)
+	s.conn.Close()
+}
+
+func (s *Server) Handle(handler report.Handler) {
+	s.handler = handler
+}
+
+func (s *Server) ServeMsg(msg *nl.Msg) bool {
+	b := msg.Body[genl.SizeofHeader:]
+
+	var pkt []byte
+	var seid uint64
+	var pdrid uint16
+	var action uint16
+
+	for len(b) > 0 {
+		hdr, n, err := nl.DecodeAttrHdr(b)
+		if err != nil {
+			return false
+		}
+		switch hdr.MaskedType() {
```

Comment:
@ShouheiNishi 

The latest report format is changed.
Please refer to the latest `internal/forwarder/buff` to modify it to support DLDR & USAR report.